### DISCUSSION TITLE
MARC language codes

### DISCUSSION THREAD

**Discussion by damien-git** [vufind-org/vufind#2928] [General]
VuFind's [language codes](https://github.com/vufind-org/vufind/blob/dev/import/translation_maps/language_map.properties) do not match [MARC language codes](https://www.loc.gov/marc/languages/language_code.html), or even [ISO 639-2](https://www.loc.gov/standards/iso639-2/php/code_list.php). VuFind is using these codes for MARC records and some XSL stylesheets; is there a particular reason to not use MARC language codes ?

**Comment by demiankatz** [comment]
@damien-git, I honestly don't know the full history of VuFind's language codes -- I believe these have been in use with only very minor changes since the project began in 2007, which is before my involvement.

There is discussion in #413 about revising the way we deal with languages -- perhaps indexing as standard codes and then using the translation system to map the codes to local translations of the language names. I think this is a good idea, but it requires us to standardize on something and figure out how to map all values to that standard, which is a difficult problem. Because my local institution's language-related needs are minimal, I am not able to devote much time to trying to solve the problem (and I'm not the best person to do so, because I have no real-life use cases). I'd welcome effort by others who are interested in improving this, though!

**Comment by damien-git** [comment]
Thanks. We have a request to use the MARC list of codes. Not much concern for translations, as most of our users are anglophones. From your reply and what I see in #413, it sounds like it would be an improvement to use the MARC codes in base VuFind. I was just wondering if I should just fix it locally, or also suggest a change in VuFind. I could create a PR with a new language_map file (without any translation) if you think it is worth it despite the lack of translations.

**Reply by demiankatz** [reply]
@damien-git, maybe one approach could be to add a "marc_language_map.properties" file with the actual MARC codes as a PR. Then users could choose to adopt the pure MARC codes, or use VuFind's existing map, depending on their needs. If we can't determine a strong reason for the VuFind list, we could deprecate the legacy one and move toward the MARC language codes in future. It does appear that VuFind's list was augmented a few years ago with some ISO codes -- maybe looking into why that happened might shed light on existing use cases and possible limitations of the pure MARC list.

**Reply by damien-git** [reply]
I took a look at differences between the MARC codes and the VuFind ones. VuFind includes more codes, and there is not a single code that MARC has and that Vufind does not, except for the discontinued codes. However the titles are different. So I guess the request we received had more to do with English titles than codes. The user prefers "Greek, Ancient" and "Greek, Modern" rather than "Ancient Greek" and "Greek", as it is easier to see both together when sorted. So rather than discarding codes that VuFind supports now, it might be better to change the titles to the official MARC ones for the codes it has.

**Reply by damien-git** [reply]
As a side note, we also have some (about 20) records using the larger ISO 639-3 codes, but this is not considered a priority.

**Reply by demiankatz** [reply]
Thanks for taking the time to look into this, @damien-git. I'm not opposed to a PR to bring the file up to date with current MARC standards; it's possible that things have simply changed a little bit in the intervening years since it was created. In the long term, a solution like the one discussed in #413 would be great, because then you could make these kinds of tweaks through your local language files instead of having to reindex things... but for now, simply updating the map seems like a reasonable starting point.

**Reply by damien-git** [reply]
Ok. I started creating that merge. There are actually 2 codes in MARC codes that are not in VuFind (cnr and srp).

**Reply by demiankatz** [reply]
It's definitely worth adding those, then! Thanks!
